include/drivers/gpio.h

```diff
@@ -817,13 +823,15 @@ static inline int gpio_port_set_clr_bits_raw(struct device *port,
 /**
  * @brief Set logical level of selected output pins.
  *
- * Pins which are simultaneously selected by set_pins and clear_pins value will
- * be toggled.
+ * Note that the behavior required for pins present in both
+ * ``set_pins`` and ``clear_pins`` is to toggle the current pin
+ * setting.
  *
  * @param port Pointer to the device structure for the driver instance.
- * @param set_pins Value indicating which pins will be set to active or toggled.
- * @param clear_pins Value indicating which pins will be set to inactive or
- *                   toggled.
+ * @param set_pins Pins selected by this mask are changed to active if
+ * they were not already active.
```

Comment:
From the commit description:
> It also clarifies the expected behavior of the driver in situations
where the driver can detect that an operation would have no effect,
which is signficant for external IO extenders where an unnecessary bus
transaction can be eliminated.

This optimization can be done by the driver also without changing the description. While we mention toggle functionality in the description of the function this text doesn't make it clear that "Pins selected by this mask can be changed to inactive if a matching pin is selected by `clear_pins` parameter." I'm not saying we shouldn't change the description of `set_pins`, `clear_pins` parameters but this proposal doesn't seem like an improvement.